stages/org.osbuild.rhsm.facts

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+"""
+The `org.osbuild.rhsm.facts` stage is allows for storing rhsm-facts format
+artifacts in the built image. Keys and values are supplied in options.
+"""
+
+import json
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"definitions": {
+  "key": {
+    "type": "string"
+  },
+  "value": {
+    "type": "string"
+  }
+},
+"properties": {
+  "facts": {
+    "type": "array",
+    "description": "Array of key/value facts.",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "key": {
+          "$ref": "#/definitions/key"
+        },
+        "value": {
+          "$ref": "#/definitions/value"
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    path = os.path.join(tree, "usr", "share", "osbuild", "self")
+    file = os.path.join(path, "rhsm.facts")
+
+    os.makedirs(path, exist_ok=True)
+
+    with open(file, "w", encoding="utf-8") as f:
```

Comment:
How about:

```suggestion
    with open(file, "x", encoding="utf-8") as f:
```

This would make this stage fail if the file exists. Note that the current code would silently drop any previous contents. This would catch manifest errors where the stage is specified multiple times, and thus some items might not end up in the resulting image?